extensions/import/src/dialogs/derivedColumnDialog.ts

```diff
@@ -0,0 +1,376 @@
+
+			this._transformationContainer.addItem(this._headerInstructionText);
+			this._transformationContainer.addItem(this._bodyInstructionText);
+
+
+			const specifyDerivedColNameContainer = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'column',
+				width: '150px'
+			}).component();
+			specifyDerivedColNameContainer.addItem(specifyDerivedColNameTable);
+
+			const flexGrid = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'row',
+				height: '800px',
+			}).component();
+			flexGrid.addItem(columnContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'overflow-y': 'auto',
+					'padding-right': '10px',
+					'height': '100vh'
+				}
+			});
+			flexGrid.addItem(this._transformationContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'overflow-y': 'auto',
+					'padding-right': '10px',
+					'height': '100vh'
+				}
+			});
+			flexGrid.addItem(specifyDerivedColNameContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'margin-left': '10px',
+					'height': '100vh'
+				}
+			});
+			const formBuilder = view.modelBuilder.formContainer().withFormItems(
+				[
+					{
+						component: flexGrid
+					}
+				],
+
+				{
+					horizontal: false
+				}
+			);
+			const form = formBuilder.withLayout({ width: '100%' }).component();
+			return view.initializeModel(form);
+		});
+
+		this._dialogObject.okButton.onClick(e => {
+			this._doneEmitter.emit('done');
+		});
+
+		this._dialogObject.cancelButton.onClick(e => {
+			this._doneEmitter.emit('close');
+		});
+
+		this._dialogObject.content = [tab];
+		azdata.window.openDialog(this._dialogObject);
+		return new Promise((resolve) => {
+			this._doneEmitter.once('done', async () => {
+				await this._provider.sendSaveTransformationRequest({
+					derivedColumnName: this.currentDerivedColumnName
+				});
+				this._model.transPreviews.push(this.currentTransformation);
+				this._model.derivedColumnName = this.currentDerivedColumnName;
+				this._model.proseColumns.push({
+					columnName: this.currentDerivedColumnName,
+					dataType: 'nvarchar(MAX)',
+					primaryKey: false,
+					nullable: true
+				});
+				resolve(true);
+				azdata.window.closeDialog(this._dialogObject);
+			});
+
+			this._doneEmitter.once('close', async () => {
+				resolve(false);
+				azdata.window.closeDialog(this._dialogObject);
+			});
+		});
+	}
+
+	private clearAndAddTransformationContainerComponents(addTable: boolean): void {
+		this._transformationContainer.updateCssStyles({
+			'width': '700px'
+		});
+		this._transformationContainer.clearItems();
+		if (addTable) {
+			this._transformationContainer.addItem(this._transformationTable);
+		}
+		else {
+			this._transformationContainer.addItem(this._headerInstructionText);
+			this._transformationContainer.addItem(this._bodyInstructionText);
+		}
+	}
+
+	private validatePage(): void {
+		if (this.currentTransformation.length > 0 && this.currentDerivedColumnName.length > 0) {
+			this._dialogObject.okButton.enabled = true;
```

Comment:
Generally we should be using withValidation to control this. Any reason you're not able to do that?